File: backend/src/main/java/com/ecommerce/backend/dto/StockCheckResponse.java
package com.ecommerce.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StockCheckResponse {
    private Long variantId;
    private int availableStock;
    private boolean inStock;
    private boolean isLowStock; // less than 10
    private String stockStatus; // "IN_STOCK", "LOW_STOCK", "OUT_OF_STOCK"
}

File: frontend_client_mobile/lib/models/stock_check_response.dart
class StockCheckResponse {
  final int variantId;
  final int availableStock;
  final bool inStock;
  final bool isLowStock;
  final String stockStatus; // "IN_STOCK", "LOW_STOCK", "OUT_OF_STOCK"

  StockCheckResponse({
    required this.variantId,
    required this.availableStock,
    required this.inStock,
    required this.isLowStock,
    required this.stockStatus,
  });

  factory StockCheckResponse.fromJson(Map<String, dynamic> json) {
    return StockCheckResponse(
      variantId: json['variantId'] as int? ?? 0,
      availableStock: json['availableStock'] as int? ?? 0,
      inStock: json['inStock'] as bool? ?? false,
      isLowStock: json['isLowStock'] as bool? ?? false,
      stockStatus: json['stockStatus'] as String? ?? 'OUT_OF_STOCK',
    );
  }
}
